jbrowse/src/org/labkey/jbrowse/JBrowseLuceneSearch.java

```diff
@@ -75,6 +79,7 @@
 public class JBrowseLuceneSearch
 {
     private static final Logger _log = LogHelper.getLogger(JBrowseLuceneSearch.class, "Logger related to JBrowse/Lucene indexing and queries");
+    private static final ExecutorService SEARCH_EXECUTOR = Executors.newFixedThreadPool(JBrowseServiceImpl.get().getCoresForLuceneSearches());
```

Comment:
I think this direction is right, but my primary concern is whether there is a more appropriate way to interact with tomcat on this. Tomcat and/or the webapp probably has a thread pool too...i wonder if it would be better practice to somehow hook into that. I will do some looking within the labkey code - could you do a little research on tomcat in general to see if there's best-practice?